src/Parsers/ParserCreateQuery.cpp

```diff
@@ -549,12 +572,24 @@ bool ParserCreateTableQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expe
 
         if (!storage_parse_result && !is_temporary)
         {
-            if (!s_as.ignore(pos, expected))
-                return false;
-            if (!table_function_p.parse(pos, as_table_function, expected))
+            if (s_as.ignore(pos, expected) && !table_function_p.parse(pos, as_table_function, expected))
             {
                 return false;
             }
+            else
+            {
+            // ENGINE can be omitted if default_table_engine is set.
+            // Need to check in Interpreter
+                if (columns_list)
+                {
+                    auto columns = columns_list->as<ASTColumns &>();
+                    if (columns.primary_key)
+                    {
+                        auto storage_ast = std::make_shared<ASTStorage>();
+                        storage = storage_ast;
+                    }
+                }
```

Comment:
I think we can add empty `ASTStorage` unconditionally and that throw exception if primary key is not specified

Reply:
I'll check this

Reply:
Done